ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -490,19 +493,33 @@ def get_ovs_br_ports(bridge):
         out = VMTopology.cmd('ovs-vsctl list-ports %s' % bridge)
         return set(out.split('\n'))
 
+
     @staticmethod
-    def get_ovs_port_bindings(bridge):
-        out = VMTopology.cmd('ovs-ofctl show %s' % bridge)
-        lines = out.split('\n')
-        result = {}
-        for line in lines:
-            matched = re.match(r'^\s+(\S+)\((\S+)\):\s+addr:.+$', line)
-            if matched:
-                port_id = matched.group(1)
-                iface_name = matched.group(2)
-                result[iface_name] = port_id
-
-        return result
+    def get_ovs_port_bindings(bridge, vlan_iface = None):
+        # Vlan interface addition may take few secs to reflect in OVS Command,
+        # Let`s retry few times in that case.
+        retries = 0
+        vlan_iface_id = None
+        while retries < RETRIES:
```

Comment:
I'd better to use `for retries in xrange(RETRIES):`

Reply:
I will change this to range. range takes more space than xrange but it is supported in python3 as well.